### PR TITLE
[HttpClient] Add withOptions() to HttplugClient and Psr18Client

### LINKED ISSUE #17382: [HttpClient] Add `withOptions()` to `HttplugClient` and `Psr18Client`
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#47936
| PR author(s) | @nicolas-grekas
| Merged in    | 6.2

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Resolves https://github.com/symfony/symfony-docs/issues/17382

### CODE REVIEW

File: http_client.rst [hunk 1]
```diff
@@ -1435,6 +1435,27 @@ Now you can make HTTP requests with the PSR-18 client as follows:
...
+            ],
+        ]);
+
+    $request = $client->createRequest('GET', '/versions.json');
+
+    // ...
+
+.. versionadded:: 6.2
+
+    The ``Psr18Client::withOptions`` method was introduced in Symfony 6.2.
```
Comment (OskarStark) [inline http_client.rst]:
```suggestion
    The ``Psr18Client::withOptions()`` method was introduced in Symfony 6.2.
```

File: http_client.rst [hunk 2]
```diff
@@ -1435,6 +1435,27 @@ Now you can make HTTP requests with the PSR-18 client as follows:
 
         $content = json_decode($response->getBody()->getContents(), true);
 
+You can also pass a set of default options to your client thanks to the
+``Psr18Client::withOptions`` method::
```
Comment (OskarStark) [inline http_client.rst]:
```suggestion
``Psr18Client::withOptions()`` method::
```

File: http_client.rst [hunk 3]
```diff
@@ -1523,6 +1544,24 @@ Then you're ready to go::
     // wait for all remaining promises to resolve
     $httpClient->wait();
 
+You can also pass a set of default options to your client thanks to the
+``HttplugClient::withOptions`` method::
```
Comment (OskarStark) [inline http_client.rst]:
```suggestion
``HttplugClient::withOptions()`` method::
```

File: http_client.rst [hunk 4]
```diff
@@ -1523,6 +1544,24 @@ Then you're ready to go::
...
+        ->withOptions([
+            'base_uri' => 'https://my.api.com',
+        ]);
+    $request = $httpClient->createRequest('GET', '/');
+
+    // ...
+
+.. versionadded:: 6.2
+
+    The ``HttplugClient::withOptions`` method was introduced in Symfony 6.2.
```
Comment (OskarStark) [inline http_client.rst]:
```suggestion
    The ``HttplugClient::withOptions()`` method was introduced in Symfony 6.2.
```

**Comment by alexandre-daubois** [comment]
I should have thought of it 😄 , thank you for the review! That's updated

**Comment by javiereguiluz** [comment]
Thank you Alex.

**Pull request merged**